src/main/java/com/liakhov/mvc/app/dao/users/UserProfileDAOImpl.java

```
@@ -0,0 +1,86 @@
+package com.liakhov.mvc.app.dao.users;
+
+import com.liakhov.mvc.app.entities.Role;
+import com.liakhov.mvc.app.entities.User;
+import com.liakhov.mvc.app.entities.UserProfile;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCallback;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+
+import javax.sql.DataSource;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.List;
+
+public class UserProfileDAOImpl implements UserProfileDAO {
+
+    private JdbcTemplate jdbcTemplate;
+    private SimpleJdbcInsert insertUser;
+
+    public void setDataSource(DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+    }
+
+    @Override
+    public void addUserProfile(UserProfile profile) {
+        String insert = "INSERT INTO user_profile (first_name, last_name, middle_name, id_user, add_date, about, e_mail, phone_number) " +
+                            " VALUES(?, ?, ?, ?, ?, ?, ?, ?)";
+        jdbcTemplate.execute(insert,
+                (PreparedStatementCallback<Boolean>) ps -> {
+                        ps.setString(1, profile.getFirstName());
+                        ps.setString(2, profile.getLastName());
+                        ps.setString(3, profile.getMiddleName());
+                        ps.setLong(4, profile.getUser().getId());
+                        ps.setLong(5, new Date().getTime());
+                        ps.setString(6, profile.getAbout());
+                        ps.setString(7, profile.geteMail());
+                        ps.setString(8, profile.getPhoneNumber());
+                        return ps.execute();
+                    });
+    }
+
+
+
+    @Override
+    public void deleteUserProfile(Integer id) {
+
```

it's better to throw `UnsupportedOperationException` in not implemented methods, so if you forget that mathod is not implemented you wont wonder why it is not working.